Cross-compilation fails for 32-bit ARM targets when the uring feature is enabled

### Describe the release item

Cross-compiling Zenoh for 32-bit ARM targets fails with the following error:
```
error: The prebuilt `sys.rs` may not be compatible with your target,
please use bindgen feature to generate new `sys.rs` of your arch
or use `--cfg=io_uring_skip_arch_check` to skip the check.
```

The failure originates from the io-uring dependency. Zenoh declares io-uring = "0.7.10" in the workspace, which currently resolves to 0.7.13 in Cargo.lock.

The affected targets are:

- arm-unknown-linux-gnueabi
- arm-unknown-linux-gnueabihf
- armv7-unknown-linux-gnueabihf

io-uring provides prebuilt bindings only for selected architectures, including x86_64 and aarch64. 32-bit ARM is not included, so the dependency intentionally stops compilation unless bindings are generated or the check is explicitly bypassed.

Zenoh's uring support is implemented through the zenoh-uring crate and is enabled by the uring feature.

## Reproduction

Using the existing Cross targets, build Zenoh with the uring feature:

```
cross build \
  --target arm-unknown-linux-gnueabi \
  --features uring
```

The build fails while compiling io-uring.

## Workaround

Adding the following to Cross.toml allows the build to complete:
```
[target.arm-unknown-linux-gnueabi.env]
passthrough = ["RUSTFLAGS=--cfg=io_uring_skip_arch_check"]
```

The same may be required for the other 32-bit ARM targets.

This bypasses the compile-time architecture check and uses the existing prebuilt bindings. However, it does not verify that those bindings are ABI-compatible with the target.

## Potential fixes
1. Enable the bindgen feature for io-uring when building for unsupported architectures.
This would generate sys.rs from the target's Linux headers during compilation. It would require the Cross images to provide:
- clang/libclang
- Appropriate target Linux headers
- A correctly configured target sysroot
2. Pass --cfg=io_uring_skip_arch_check for the affected targets.
This is simpler and avoids additional build dependencies, but it suppresses a compatibility check and may conceal incorrect architecture-specific bindings.
3. Disable the uring feature for targets that do not support or require io_uring.

## Evidence

The failure is reproduced in this GitHub Actions run:
https://github.com/eclipse-zenoh/zenoh/actions/runs/29709345081/job/88251228067

The job fails while cross-compiling Zenoh for a 32-bit ARM target with the io-uring architecture compatibility error described above.

I tried to build against the newer images and enabling bindgen for io-uring using the following patch:

```
diff --git i/Cross.toml w/Cross.toml
index 8f0ad3160..9d2603682 100644
--- i/Cross.toml
+++ w/Cross.toml
@@ -2,7 +2,14 @@
 image = "jenoch/rust-cross:x86_64-unknown-linux-musl"
 
 [target.arm-unknown-linux-gnueabi]
-image = "jenoch/rust-cross:arm-unknown-linux-gnueabi"
+pre-build = [
+    "apt-get update",
+    "apt-get install linux-libc-dev-armel-cross"
+]
+image = "ghcr.io/cross-rs/arm-unknown-linux-gnueabi@sha256:a6ff224988e555af0e3dc193527aa0721be6ee8d5eaf44aed4c3cbe71d3e6996"
+
+[target.arm-unknown-linux-gnueabi.env]
+passthrough = ["BINDGEN_EXTRA_CLANG_ARGS=--target=arm-linux-gnueabi --sysroot=/usr/arm-linux-gnueabi"]
 
 [target.arm-unknown-linux-gnueabihf]
 image = "jenoch/rust-cross:arm-unknown-linux-gnueabihf"
diff --git i/commons/zenoh-uring/Cargo.toml w/commons/zenoh-uring/Cargo.toml
index 83a117e95..10ab4395c 100644
--- i/commons/zenoh-uring/Cargo.toml
+++ w/commons/zenoh-uring/Cargo.toml
@@ -30,7 +30,7 @@ uring_trace = []
 [target.'cfg(target_os = "linux")'.dependencies]
 atomic-queue = { workspace = true }
 flume = { workspace = true }
-io-uring = { workspace = true }
+io-uring = { workspace = true, features = ["bindgen"] }
 libc = { workspace = true }
 nix = { workspace = true, features = ["event"] }
 rand = { workspace = true, features = ["default"] }
```

The Cross image provides linux-libc-dev-armel-cross version 6.8.0-25.25cross1, with the expected target header at /usr/arm-linux-gnueabi/include/linux/io_uring.h. However, that header lacks newer definitions required by io-uring 0.7.13, including IORING_OP_BIND, IORING_REGISTER_NAPI, and IORING_ZCRX_*. Enabling bindgen therefore generates incomplete bindings and the dependency fails with 47 compilation errors.

```
$ DOCKER_DEFAULT_PLATFORM=linux/amd64 CROSS_CONTAINER_OPTS="--platform=linux/amd64" cross build --target arm-unknown-linux-gnueabi --features uring
[+] Building 0.0s (6/6) FINISHED                                         docker:desktop-linux
 => [internal] load build definition from Dockerfile.arm-unknown-linux-gnueabi-custom    0.0s
 => => transferring dockerfile: 314B                                                     0.0s
 => [internal] load metadata for ghcr.io/cross-rs/arm-unknown-linux-gnueabi@sha256:a6ff  0.0s
 => [internal] load .dockerignore                                                        0.0s
 => => transferring context: 54B                                                         0.0s
 => [1/2] FROM ghcr.io/cross-rs/arm-unknown-linux-gnueabi@sha256:a6ff224988e555af0e3dc1  0.0s
 => CACHED [2/2] RUN eval "apt-get update
apt-get install linux-libc-dev-armel-cross"    0.0s
 => exporting to image                                                                   0.0s
 => => exporting layers                                                                  0.0s
 => => writing image sha256:7dbfb516ef649888c07c9a1ddbdb32e4504970863ab415944605706d817  0.0s
 => => naming to docker.io/library/cross-custom-zenoh:arm-unknown-linux-gnueabi-9b60a-p  0.0s

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/mvpzfumr2fxkjvrumev63wt7n
   Compiling bindgen v0.69.5
   Compiling clang-sys v1.8.1
   Compiling vec_map v0.8.2
   Compiling futures v0.3.32
   Compiling zenoh-util v1.9.0 (/project/commons/zenoh-util)
   Compiling icu_properties v2.2.0
   Compiling icu_normalizer v2.2.0
   Compiling num-bigint v0.4.8
   Compiling block-buffer v0.10.4
   Compiling crossbeam-epoch v0.9.20
   Compiling json5 v0.4.1
   Compiling serde_spanned v1.1.1
   Compiling toml_datetime v0.7.5+spec-1.1.0
   Compiling unsafe-libyaml v0.2.11
   Compiling subtle v2.6.1
   Compiling toml_writer v1.1.2+spec-1.1.0
   Compiling zeroize v1.9.0
   Compiling winnow v0.7.15
   Compiling ryu v1.0.23
   Compiling digest v0.10.7
   Compiling validated_struct v2.2.0
   Compiling secrecy v0.8.0
   Compiling idna_adapter v1.2.2
   Compiling toml v0.9.12+spec-1.1.0
   Compiling serde_yaml v0.9.34+deprecated
   Compiling num-rational v0.4.2
   Compiling num-iter v0.1.46
   Compiling tokio-util v0.7.18
   Compiling num-complex v0.4.6
   Compiling parking_lot_core v0.9.12
   Compiling inout v0.1.4
   Compiling crossbeam-deque v0.8.7
   Compiling serde_with v3.21.0
   Compiling io-uring v0.7.13
   Compiling nonempty-collections v0.3.1
   Compiling num_cpus v1.17.0
   Compiling rustix v1.1.4
   Compiling bit-vec v0.6.3
   Compiling percent-encoding v2.3.2
   Compiling cipher v0.4.4
   Compiling zenoh-config v1.9.0 (/project/commons/zenoh-config)
   Compiling bit-set v0.5.3
   Compiling form_urlencoded v1.2.2
   Compiling num v0.4.3
   Compiling idna v1.1.0
   Compiling zenoh-codec v1.9.0 (/project/commons/zenoh-codec)
   Compiling crossbeam-channel v0.5.16
   Compiling concurrent-queue v2.5.0
   Compiling crossbeam-queue v0.3.13
   Compiling stabby-abi v72.1.8
   Compiling derive_more-impl v2.1.1
   Compiling socket2 v0.5.10
   Compiling nom v8.0.0
   Compiling vsimd v0.8.0
   Compiling heck v0.5.0
error[E0432]: unresolved imports `sys::io_uring_region_desc`, `sys::io_uring_zcrx_area_reg`, `sys::io_uring_zcrx_cqe`, `sys::io_uring_zcrx_ifq_reg`, `sys::io_uring_zcrx_rqe`, `sys::IORING_MEM_REGION_TYPE_USER`, `sys::IORING_ZCRX_AREA_SHIFT`, `sys::IOU_PBUF_RING_INC`, `sys::IOU_PBUF_RING_MMAP`
  --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:54:5
   |
54 |     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
   |     ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^ no `io_uring_zcrx_ifq_reg` in `sys`
   |     |                     |                       |
   |     |                     |                       no `io_uring_zcrx_cqe` in `sys`
   |     |                     no `io_uring_zcrx_area_reg` in `sys`
   |     no `io_uring_region_desc` in `sys`
55 |     io_uring_zcrx_rqe, IORING_MEM_REGION_TYPE_USER, IORING_ZCRX_AREA_SHIFT, IOU_PBUF_RING_INC,
   |     ^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^ no `IOU_PBUF_RING_INC` in `sys`
   |     |                  |                            |
   |     |                  |                            no `IORING_ZCRX_AREA_SHIFT` in `sys`
   |     |                  no `IORING_MEM_REGION_TYPE_USER` in `sys`
   |     no `io_uring_zcrx_rqe` in `sys`
56 |     IOU_PBUF_RING_MMAP,
   |     ^^^^^^^^^^^^^^^^^^ no `IOU_PBUF_RING_MMAP` in `sys`
   |
help: a similar name exists in the module
   |
54 -     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
54 +     io_uring_register, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
   |
help: a similar name exists in the module
   |
54 -     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
54 +     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_cqe, io_uring_zcrx_ifq_reg,
   |
help: a similar name exists in the module
   |
54 -     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
54 +     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_buf_reg,
   |
help: a similar name exists in the module
   |
55 -     io_uring_zcrx_rqe, IORING_MEM_REGION_TYPE_USER, IORING_ZCRX_AREA_SHIFT, IOU_PBUF_RING_INC,
55 +     io_uring_cqe, IORING_MEM_REGION_TYPE_USER, IORING_ZCRX_AREA_SHIFT, IOU_PBUF_RING_INC,
   |

   Compiling static_assertions v1.1.0
   Compiling linux-raw-sys v0.12.1
error[E0425]: cannot find value `IORING_CQE_F_BUF_MORE` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/cqueue.rs:305:18
    |
305 |     flags & sys::IORING_CQE_F_BUF_MORE != 0
    |                  ^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:129:1
    |
129 | pub const IORING_CQE_F_BUFFER: u32 = 1;
    | --------------------------------------- similarly named constant `IORING_CQE_F_BUFFER` defined here
    |
help: a constant with a similar name exists
    |
305 -     flags & sys::IORING_CQE_F_BUF_MORE != 0
305 +     flags & sys::IORING_CQE_F_BUFFER != 0
    |

error[E0425]: cannot find value `SOCKET_URING_OP_SETSOCKOPT` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:664:61
    |
664 |         sqe.__bindgen_anon_1.__bindgen_anon_1.cmd_op = sys::SOCKET_URING_OP_SETSOCKOPT;
    |                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_OP_FTRUNCATE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2105:27
     |
2105 |     pub const CODE = sys::IORING_OP_FTRUNCATE;
     |                           ^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1133:1
     |
1133 | pub const IORING_OP_FALLOCATE: io_uring_op = 17;
     | ------------------------------------------------ similarly named constant `IORING_OP_FALLOCATE` defined here
     |
help: a constant with a similar name exists
     |
2105 -     pub const CODE = sys::IORING_OP_FTRUNCATE;
2105 +     pub const CODE = sys::IORING_OP_FALLOCATE;
     |

error[E0425]: cannot find value `IORING_RECVSEND_BUNDLE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2140:28
     |
2140 |         sqe.ioprio |= sys::IORING_RECVSEND_BUNDLE as u16;
     |                            ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_RECVSEND_BUNDLE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2175:28
     |
2175 |         sqe.ioprio |= sys::IORING_RECVSEND_BUNDLE as u16;
     |                            ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_RECVSEND_BUNDLE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2217:28
     |
2217 |         sqe.ioprio |= sys::IORING_RECVSEND_BUNDLE as u16;
     |                            ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_OP_BIND` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2233:27
     |
2233 |     pub const CODE = sys::IORING_OP_BIND;
     |                           ^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1142:1
     |
1142 | pub const IORING_OP_SEND: io_uring_op = 26;
     | ------------------------------------------- similarly named constant `IORING_OP_SEND` defined here
     |
help: a constant with a similar name exists
     |
2233 -     pub const CODE = sys::IORING_OP_BIND;
2233 +     pub const CODE = sys::IORING_OP_SEND;
     |

error[E0425]: cannot find value `IORING_OP_LISTEN` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2255:27
     |
2255 |     pub const CODE = sys::IORING_OP_LISTEN;
     |                           ^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1171:1
     |
1171 | pub const IORING_OP_LAST: io_uring_op = 55;
     | ------------------------------------------- similarly named constant `IORING_OP_LAST` defined here
     |
help: a constant with a similar name exists
     |
2255 -     pub const CODE = sys::IORING_OP_LISTEN;
2255 +     pub const CODE = sys::IORING_OP_LAST;
     |

error[E0425]: cannot find value `IORING_OP_RECV_ZC` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2280:27
     |
2280 |     pub const CODE = sys::IORING_OP_RECV_ZC;
     |                           ^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1143:1
     |
1143 | pub const IORING_OP_RECV: io_uring_op = 27;
     | ------------------------------------------- similarly named constant `IORING_OP_RECV` defined here
     |
help: a constant with a similar name exists
     |
2280 -     pub const CODE = sys::IORING_OP_RECV_ZC;
2280 +     pub const CODE = sys::IORING_OP_RECV;
     |

error[E0425]: cannot find value `IORING_OP_EPOLL_WAIT` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2305:27
     |
2305 |     pub const CODE = sys::IORING_OP_EPOLL_WAIT;
     |                           ^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1145:1
     |
1145 | pub const IORING_OP_EPOLL_CTL: io_uring_op = 29;
     | ------------------------------------------------ similarly named constant `IORING_OP_EPOLL_CTL` defined here
     |
help: a constant with a similar name exists
     |
2305 -     pub const CODE = sys::IORING_OP_EPOLL_WAIT;
2305 +     pub const CODE = sys::IORING_OP_EPOLL_CTL;
     |

error[E0425]: cannot find value `IORING_OP_READV_FIXED` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2333:27
     |
2333 |     pub const CODE = sys::IORING_OP_READV_FIXED;
     |                           ^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1120:1
     |
1120 | pub const IORING_OP_READ_FIXED: io_uring_op = 4;
     | ------------------------------------------------ similarly named constant `IORING_OP_READ_FIXED` defined here
     |
help: a constant with a similar name exists
     |
2333 -     pub const CODE = sys::IORING_OP_READV_FIXED;
2333 +     pub const CODE = sys::IORING_OP_READ_FIXED;
     |

error[E0425]: cannot find value `IORING_OP_WRITEV_FIXED` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2364:27
     |
2364 |     pub const CODE = sys::IORING_OP_WRITEV_FIXED;
     |                           ^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1121:1
     |
1121 | pub const IORING_OP_WRITE_FIXED: io_uring_op = 5;
     | ------------------------------------------------- similarly named constant `IORING_OP_WRITE_FIXED` defined here
     |
help: a constant with a similar name exists
     |
2364 -     pub const CODE = sys::IORING_OP_WRITEV_FIXED;
2364 +     pub const CODE = sys::IORING_OP_WRITE_FIXED;
     |

error[E0425]: cannot find value `IORING_OP_PIPE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2393:27
     |
2393 |     pub const CODE = sys::IORING_OP_PIPE;
     |                           ^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1116:1
     |
1116 | pub const IORING_OP_NOP: io_uring_op = 0;
     | ----------------------------------------- similarly named constant `IORING_OP_NOP` defined here
     |
help: a constant with a similar name exists
     |
2393 -     pub const CODE = sys::IORING_OP_PIPE;
2393 +     pub const CODE = sys::IORING_OP_NOP;
     |

error[E0425]: cannot find value `IORING_ENTER_ABS_TIMER` in module `sys`
  --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:37:32
   |
37 |         const ABS_TIMER = sys::IORING_ENTER_ABS_TIMER;
   |                                ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_ENTER_EXT_ARG_REG` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:40:34
    |
 40 |         const EXT_ARG_REG = sys::IORING_ENTER_EXT_ARG_REG;
    |                                  ^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:146:1
    |
146 | pub const IORING_ENTER_EXT_ARG: u32 = 8;
    | ---------------------------------------- similarly named constant `IORING_ENTER_EXT_ARG` defined here
    |
help: a constant with a similar name exists
    |
 40 -         const EXT_ARG_REG = sys::IORING_ENTER_EXT_ARG_REG;
 40 +         const EXT_ARG_REG = sys::IORING_ENTER_EXT_ARG;
    |

error[E0425]: cannot find value `IORING_ENTER_NO_IOWAIT` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:43:32
    |
 43 |         const NO_IOWAIT = sys::IORING_ENTER_NO_IOWAIT;
    |                                ^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:145:1
    |
145 | pub const IORING_ENTER_SQ_WAIT: u32 = 4;
    | ---------------------------------------- similarly named constant `IORING_ENTER_SQ_WAIT` defined here
    |
help: a constant with a similar name exists
    |
 43 -         const NO_IOWAIT = sys::IORING_ENTER_NO_IOWAIT;
 43 +         const NO_IOWAIT = sys::IORING_ENTER_SQ_WAIT;
    |

error[E0422]: cannot find struct, variant or union type `io_uring_clone_buffers` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:385:24
    |
385 |         let arg = sys::io_uring_clone_buffers {
    |                        ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_REGISTER_CLONE_BUFFERS` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:395:18
     |
 395 |             sys::IORING_REGISTER_CLONE_BUFFERS,
     |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1606:1
     |
1606 | pub const IORING_REGISTER_BUFFERS: _bindgen_ty_7 = 0;
     | ----------------------------------------------------- similarly named constant `IORING_REGISTER_BUFFERS` defined here
     |
help: a constant with a similar name exists
     |
 395 -             sys::IORING_REGISTER_CLONE_BUFFERS,
 395 +             sys::IORING_REGISTER_BUFFERS,
     |

error[E0425]: cannot find type `io_uring_clone_buffers` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:396:29
    |
396 |             cast_ptr::<sys::io_uring_clone_buffers>(&arg).cast(),
    |                             ^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_REGISTER_NAPI` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:666:18
     |
 666 |             sys::IORING_REGISTER_NAPI,
     |                  ^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1633:1
     |
1633 | pub const IORING_REGISTER_LAST: _bindgen_ty_7 = 27;
     | --------------------------------------------------- similarly named constant `IORING_REGISTER_LAST` defined here
     |
help: a constant with a similar name exists
     |
 666 -             sys::IORING_REGISTER_NAPI,
 666 +             sys::IORING_REGISTER_LAST,
     |

error[E0425]: cannot find value `IORING_UNREGISTER_NAPI` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:683:18
     |
 683 |             sys::IORING_UNREGISTER_NAPI,
     |                  ^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1633:1
     |
1633 | pub const IORING_REGISTER_LAST: _bindgen_ty_7 = 27;
     | --------------------------------------------------- similarly named constant `IORING_REGISTER_LAST` defined here
     |
help: a constant with a similar name exists
     |
 683 -             sys::IORING_UNREGISTER_NAPI,
 683 +             sys::IORING_REGISTER_LAST,
     |

error[E0425]: cannot find value `IO_URING_NAPI_STATIC_ADD_ID` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:700:43
    |
700 |         self.register_napi_static_op(sys::IO_URING_NAPI_STATIC_ADD_ID as _, napi_id)
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IO_URING_NAPI_STATIC_DEL_ID` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:712:43
    |
712 |         self.register_napi_static_op(sys::IO_URING_NAPI_STATIC_DEL_ID as _, napi_id)
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0422]: cannot find struct, variant or union type `io_uring_napi` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:719:28
     |
 719 |         let mut arg = sys::io_uring_napi {
     |                            ^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1172:1
     |
1172 | pub type io_uring_op = libc::c_uint;
     | ------------------------------------ similarly named type alias `io_uring_op` defined here
     |
help: a type alias with a similar name exists
     |
 719 -         let mut arg = sys::io_uring_napi {
 719 +         let mut arg = sys::io_uring_op {
     |

error[E0425]: cannot find value `IORING_REGISTER_NAPI` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:726:18
     |
 726 |             sys::IORING_REGISTER_NAPI,
     |                  ^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1633:1
     |
1633 | pub const IORING_REGISTER_LAST: _bindgen_ty_7 = 27;
     | --------------------------------------------------- similarly named constant `IORING_REGISTER_LAST` defined here
     |
help: a constant with a similar name exists
     |
 726 -             sys::IORING_REGISTER_NAPI,
 726 +             sys::IORING_REGISTER_LAST,
     |

error[E0425]: cannot find type `io_uring_napi` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:727:36
     |
 727 |             (&mut arg as *mut sys::io_uring_napi).cast(),
     |                                    ^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1172:1
     |
1172 | pub type io_uring_op = libc::c_uint;
     | ------------------------------------ similarly named type alias `io_uring_op` defined here
     |
help: a type alias with a similar name exists
     |
 727 -             (&mut arg as *mut sys::io_uring_napi).cast(),
 727 +             (&mut arg as *mut sys::io_uring_op).cast(),
     |

error[E0425]: cannot find type `io_uring_zcrx_ifq_reg` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:875:43
     |
 875 |     pub fn register_ifq(&self, reg: &sys::io_uring_zcrx_ifq_reg) -> io::Result<()> {
     |                                           ^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:2427:1
     |
2427 | pub struct io_uring_buf_reg {
     | --------------------------- similarly named struct `io_uring_buf_reg` defined here
     |
help: a struct with a similar name exists
     |
 875 -     pub fn register_ifq(&self, reg: &sys::io_uring_zcrx_ifq_reg) -> io::Result<()> {
 875 +     pub fn register_ifq(&self, reg: &sys::io_uring_buf_reg) -> io::Result<()> {
     |

error[E0425]: cannot find value `IORING_REGISTER_ZCRX_IFQ` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:878:18
     |
 878 |             sys::IORING_REGISTER_ZCRX_IFQ,
     |                  ^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1623:1
     |
1623 | pub const IORING_REGISTER_IOWQ_AFF: _bindgen_ty_7 = 17;
     | ------------------------------------------------------- similarly named constant `IORING_REGISTER_IOWQ_AFF` defined here
     |
help: a constant with a similar name exists
     |
 878 -             sys::IORING_REGISTER_ZCRX_IFQ,
 878 +             sys::IORING_REGISTER_IOWQ_AFF,
     |

error[E0425]: cannot find type `io_uring_zcrx_ifq_reg` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/submit.rs:879:29
     |
 879 |             cast_ptr::<sys::io_uring_zcrx_ifq_reg>(reg) as _,
     |                             ^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:2427:1
     |
2427 | pub struct io_uring_buf_reg {
     | --------------------------- similarly named struct `io_uring_buf_reg` defined here
     |
help: a struct with a similar name exists
     |
 879 -             cast_ptr::<sys::io_uring_zcrx_ifq_reg>(reg) as _,
 879 +             cast_ptr::<sys::io_uring_buf_reg>(reg) as _,
     |

error[E0425]: cannot find value `IORING_REGISTER_SRC_REGISTERED` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:166:37
     |
 166 |         const SRC_REGISTERED = sys::IORING_REGISTER_SRC_REGISTERED;
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1634:1
     |
1634 | pub const IORING_REGISTER_USE_REGISTERED_RING: _bindgen_ty_7 = 2147483648;
     | -------------------------------------------------------------------------- similarly named constant `IORING_REGISTER_USE_REGISTERED_RING` defined here
     |
help: a constant with a similar name exists
     |
 166 -         const SRC_REGISTERED = sys::IORING_REGISTER_SRC_REGISTERED;
 166 +         const SRC_REGISTERED = sys::IORING_REGISTER_USE_REGISTERED_RING;
     |

error[E0425]: cannot find value `IORING_REGISTER_DST_REPLACE` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:171:34
     |
 171 |         const DST_REPLACE = sys::IORING_REGISTER_DST_REPLACE;
     |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1612:1
     |
1612 | pub const IORING_REGISTER_FILES_UPDATE: _bindgen_ty_7 = 6;
     | ---------------------------------------------------------- similarly named constant `IORING_REGISTER_FILES_UPDATE` defined here
     |
help: a constant with a similar name exists
     |
 171 -         const DST_REPLACE = sys::IORING_REGISTER_DST_REPLACE;
 171 +         const DST_REPLACE = sys::IORING_REGISTER_FILES_UPDATE;
     |

error[E0425]: cannot find value `IO_URING_NAPI_TRACKING_DYNAMIC` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:731:43
    |
731 |             NapiTracking::Dynamic => sys::IO_URING_NAPI_TRACKING_DYNAMIC,
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IO_URING_NAPI_TRACKING_STATIC` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:732:42
    |
732 |             NapiTracking::Static => sys::IO_URING_NAPI_TRACKING_STATIC,
    |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IO_URING_NAPI_TRACKING_INACTIVE` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:733:44
    |
733 |             NapiTracking::Inactive => sys::IO_URING_NAPI_TRACKING_INACTIVE,
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find type `io_uring_napi` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:749:22
     |
 749 | pub struct Napi(sys::io_uring_napi);
     |                      ^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1172:1
     |
1172 | pub type io_uring_op = libc::c_uint;
     | ------------------------------------ similarly named type alias `io_uring_op` defined here
     |
help: a type alias with a similar name exists
     |
 749 - pub struct Napi(sys::io_uring_napi);
 749 + pub struct Napi(sys::io_uring_op);
     |

error[E0422]: cannot find struct, variant or union type `io_uring_napi` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:753:19
     |
 753 |         Napi(sys::io_uring_napi {
     |                   ^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1172:1
     |
1172 | pub type io_uring_op = libc::c_uint;
     | ------------------------------------ similarly named type alias `io_uring_op` defined here
     |
help: a type alias with a similar name exists
     |
 753 -         Napi(sys::io_uring_napi {
 753 +         Napi(sys::io_uring_op {
     |

error[E0425]: cannot find value `IO_URING_NAPI_REGISTER_OP` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:756:26
    |
756 |             opcode: sys::IO_URING_NAPI_REGISTER_OP as _,
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IO_URING_NAPI_TRACKING_DYNAMIC` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:758:28
    |
758 |             op_param: sys::IO_URING_NAPI_TRACKING_DYNAMIC,
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find type `io_uring_napi` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:791:54
     |
 791 |     pub(crate) fn as_mut_ptr(&mut self) -> *mut sys::io_uring_napi {
     |                                                      ^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1172:1
     |
1172 | pub type io_uring_op = libc::c_uint;
     | ------------------------------------ similarly named type alias `io_uring_op` defined here
     |
help: a type alias with a similar name exists
     |
 791 -     pub(crate) fn as_mut_ptr(&mut self) -> *mut sys::io_uring_napi {
 791 +     pub(crate) fn as_mut_ptr(&mut self) -> *mut sys::io_uring_op {
     |

error[E0425]: cannot find value `IORING_FEAT_RECVSEND_BUNDLE` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/lib.rs:646:32
    |
646 |         self.0.features & sys::IORING_FEAT_RECVSEND_BUNDLE != 0
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `sys`

error[E0425]: cannot find value `IORING_FEAT_MIN_TIMEOUT` in module `sys`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/lib.rs:654:32
     |
 654 |         self.0.features & sys::IORING_FEAT_MIN_TIMEOUT != 0
     |                                ^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:1131:1
     |
1131 | pub const IORING_OP_LINK_TIMEOUT: io_uring_op = 15;
     | --------------------------------------------------- similarly named constant `IORING_OP_LINK_TIMEOUT` defined here
     |
help: a constant with a similar name exists
     |
 654 -         self.0.features & sys::IORING_FEAT_MIN_TIMEOUT != 0
 654 +         self.0.features & sys::IORING_OP_LINK_TIMEOUT != 0
     |

error[E0425]: cannot find value `IORING_FEAT_NO_IOWAIT` in module `sys`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/lib.rs:666:32
    |
666 |         self.0.features & sys::IORING_FEAT_NO_IOWAIT != 0
    |                                ^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /target/arm-unknown-linux-gnueabi/debug/build/io-uring-a3a37f5df69e7de7/out/sys.rs:149:1
    |
149 | pub const IORING_FEAT_NODROP: u32 = 2;
    | -------------------------------------- similarly named constant `IORING_FEAT_NODROP` defined here
    |
help: a constant with a similar name exists
    |
666 -         self.0.features & sys::IORING_FEAT_NO_IOWAIT != 0
666 +         self.0.features & sys::IORING_FEAT_NODROP != 0
    |

   Compiling outref v0.5.2
   Compiling parking v2.2.1
   Compiling getrandom v0.4.3
   Compiling uuid v1.24.0
error[E0609]: no field `zcrx_ifq_idx` on type `io_uring_sqe__bindgen_ty_5`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2290:30
     |
2290 |         sqe.__bindgen_anon_5.zcrx_ifq_idx = ifq;
     |                              ^^^^^^^^^^^^ unknown field
     |
     = note: available fields are: `splice_fd_in`, `file_index`, `optlen`, `__bindgen_anon_1`

error[E0609]: no field `pipe_flags` on type `io_uring_sqe__bindgen_ty_3`
    --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/opcode.rs:2402:30
     |
2402 |         sqe.__bindgen_anon_3.pipe_flags = flags;
     |                              ^^^^^^^^^^ unknown field
     |
help: a field with a similar name exists
     |
2402 -         sqe.__bindgen_anon_3.pipe_flags = flags;
2402 +         sqe.__bindgen_anon_3.open_flags = flags;
     |

error[E0560]: struct `io_uring_getevents_arg` has no field named `min_wait_usec`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:273:13
    |
273 |             min_wait_usec: 0,
    |             ^^^^^^^^^^^^^ `io_uring_getevents_arg` does not have this field
    |
    = note: available fields are: `pad`

error[E0609]: no field `min_wait_usec` on type `io_uring_getevents_arg`
   --> /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/io-uring-0.7.13/src/types.rs:311:19
    |
311 |         self.args.min_wait_usec = min_wait_usec;
    |                   ^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sigmask`, `sigmask_sz`, `pad`, `ts`

Some errors have detailed explanations: E0422, E0425, E0432, E0560, E0609.
For more information about an error, try `rustc --explain E0422`.
error: could not compile `io-uring` (lib) due to 47 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/commons/zenoh-uring/Cargo.toml
+++ b/commons/zenoh-uring/Cargo.toml
@@ -27,7 +27,9 @@ version = { workspace = true }
 [features]
 uring_trace = []
 
-[target.'cfg(target_os = "linux")'.dependencies]
+# Build the code in this crate only on platforms supported by io-uring.
+# Listed from https://github.com/tokio-rs/io-uring/blob/a52c1c07ffa7ab64750fd6a39f2f0a922f205815/src/sys/mod.rs#L18-L28
+[target.'cfg(all(target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))'.dependencies]
 atomic-queue = { workspace = true }
 flume = { workspace = true }
 io-uring = { workspace = true }

--- a/commons/zenoh-uring/src/lib.rs
+++ b/commons/zenoh-uring/src/lib.rs
@@ -17,7 +17,7 @@
 //! This module is intended for Zenoh's internal use.
 //!
 //! [Click here for Zenoh's documentation](https://docs.rs/zenoh/latest/zenoh)
-#[cfg(target_os = "linux")]
+#[cfg(all(target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 mod linux;
-#[cfg(target_os = "linux")]
+#[cfg(all(target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 pub use linux::*;

--- a/commons/zenoh-uring/src/lib.rs
+++ b/commons/zenoh-uring/src/lib.rs
@@ -17,7 +17,25 @@
 //! This module is intended for Zenoh's internal use.
 //!
 //! [Click here for Zenoh's documentation](https://docs.rs/zenoh/latest/zenoh)
-#[cfg(all(target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 mod linux;
-#[cfg(all(target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 pub use linux::*;

--- a/io/zenoh-transport/src/common/batch.rs
+++ b/io/zenoh-transport/src/common/batch.rs
@@ -462,7 +462,17 @@ impl<TBuffer: BacktrackableReader + Buffer> RBatch<TBuffer> {
     }
 }
 
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 impl<TBuffer: BacktrackableReader + Buffer> RBatch<TBuffer>
 where
     RBatch<TBuffer>: DecompressUring,

--- a/io/zenoh-transport/src/common/batch.rs
+++ b/io/zenoh-transport/src/common/batch.rs
@@ -462,7 +462,7 @@ impl<TBuffer: BacktrackableReader + Buffer> RBatch<TBuffer> {
     }
 }
 
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 impl<TBuffer: BacktrackableReader + Buffer> RBatch<TBuffer>
 where
     RBatch<TBuffer>: DecompressUring,

--- a/io/zenoh-transport/src/lib.rs
+++ b/io/zenoh-transport/src/lib.rs
@@ -26,7 +26,17 @@ pub mod unicast;
 pub mod shm;
 #[cfg(feature = "shared-memory")]
 mod shm_context;
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 mod uring;
 
 use std::{any::Any, sync::Arc};

--- a/io/zenoh-transport/src/lib.rs
+++ b/io/zenoh-transport/src/lib.rs
@@ -26,7 +26,7 @@ pub mod unicast;
 pub mod shm;
 #[cfg(feature = "shared-memory")]
 mod shm_context;
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 mod uring;
 
 use std::{any::Any, sync::Arc};

--- a/io/zenoh-transport/src/manager.rs
+++ b/io/zenoh-transport/src/manager.rs
@@ -38,7 +38,7 @@ use super::{
 };
 #[cfg(feature = "shared-memory")]
 use crate::shm_context::ShmContext;
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 use crate::uring::Uring;
 use crate::{
     multicast::manager::{
@@ -175,7 +175,7 @@ pub struct TransportManagerState {
     pub multicast: TransportManagerStateMulticast,
     #[cfg(feature = "shared-memory")]
     pub shm_context: Option<ShmContext>,
-    #[cfg(all(feature = "uring", target_os = "linux"))]
+    #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
     pub uring: Option<Uring>,
 }
 
@@ -498,12 +498,30 @@ impl TransportManagerBuilder {
             region_name: self.region_name,
         };
 
+        if cfg!(feature = "uring")
+            && !cfg!(all(
+                target_os = "linux",
+                any(
+                    target_arch = "x86_64",
+                    target_arch = "aarch64",
+                    target_arch = "riscv64",
+                    target_arch = "loongarch64",
+                    target_arch = "powerpc64"
+                )
+            ))
+        {
+            tracing::warn!(
+                "The `uring` feature is enabled, but io_uring is only supported with Linux on x86_64, \
+                 aarch64, riscv64, loongarch64 or powerpc64; falling back to tokio RX."
+            );
+        }
+
         let state = TransportManagerState {
             unicast: unicast.state,
             multicast: multicast.state,
             #[cfg(feature = "shared-memory")]
             shm_context,
-            #[cfg(all(feature = "uring", target_os = "linux"))]
+            #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
             uring: Uring::new(config.batch_size as usize, config.link_rx_buffer_size)
                 .map_err(|e| {
                     tracing::warn!("io_uring reactor init failed, falling back to tokio RX: {e}");

--- a/io/zenoh-transport/src/manager.rs
+++ b/io/zenoh-transport/src/manager.rs
@@ -38,7 +38,17 @@ use super::{
 };
 #[cfg(feature = "shared-memory")]
 use crate::shm_context::ShmContext;
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 use crate::uring::Uring;
 use crate::{
     multicast::manager::{
@@ -175,7 +185,17 @@ pub struct TransportManagerState {
     pub multicast: TransportManagerStateMulticast,
     #[cfg(feature = "shared-memory")]
     pub shm_context: Option<ShmContext>,
-    #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+    #[cfg(all(
+        feature = "uring",
+        target_os = "linux",
+        any(
+            target_arch = "x86_64",
+            target_arch = "aarch64",
+            target_arch = "riscv64",
+            target_arch = "loongarch64",
+            target_arch = "powerpc64"
+        )
+    ))]
     pub uring: Option<Uring>,
 }
 
@@ -521,7 +541,17 @@ impl TransportManagerBuilder {
             multicast: multicast.state,
             #[cfg(feature = "shared-memory")]
             shm_context,
-            #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+            #[cfg(all(
+                feature = "uring",
+                target_os = "linux",
+                any(
+                    target_arch = "x86_64",
+                    target_arch = "aarch64",
+                    target_arch = "riscv64",
+                    target_arch = "loongarch64",
+                    target_arch = "powerpc64"
+                )
+            ))]
             uring: Uring::new(config.batch_size as usize, config.link_rx_buffer_size)
                 .map_err(|e| {
                     tracing::warn!("io_uring reactor init failed, falling back to tokio RX: {e}");

--- a/io/zenoh-transport/src/unicast/universal/link.rs
+++ b/io/zenoh-transport/src/unicast/universal/link.rs
@@ -1,4 +1,4 @@
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 use std::fmt::Debug;
 //
 // Copyright (c) 2023 ZettaScale Technology
@@ -27,7 +27,7 @@ use futures::{future::select_all, task::AtomicWaker};
 use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 use zenoh_buffers::ZSlice;
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 use zenoh_buffers::{
     buffer::Buffer,
     reader::{BacktrackableReader, HasReader},
@@ -42,7 +42,7 @@ use zenoh_sync::RecyclingObjectPool;
 #[cfg(feature = "unstable")]
 use zenoh_sync::{event, Notifier, Waiter};
 use zenoh_task::TaskController;
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 use zenoh_uring::api::reader::{
     fragmented_batch::{DefragmentationState, FragmentedBatch},
     rx_buffer::RxBuffer,
@@ -370,7 +370,7 @@ async fn rx_task(
     cancellation_token: CancellationToken,
     #[cfg(feature = "stats")] stats: zenoh_stats::LinkStats,
 ) -> ZResult<()> {
-    #[cfg(all(feature = "uring", target_os = "linux"))]
+    #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
     if transport.manager.state.uring.is_some() && link.link.get_fd().is_ok() {
         return rx_task_uring(
             link,
@@ -571,7 +571,7 @@ impl Drop for TimeoutTracker {
     }
 }
 
-#[cfg(all(feature = "uring", target_os = "linux"))]
+#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
 async fn rx_task_uring(
     link: &mut TransportLinkUnicastRx,
     transport: TransportUnicastUniversal,

--- a/io/zenoh-transport/src/unicast/universal/link.rs
+++ b/io/zenoh-transport/src/unicast/universal/link.rs
@@ -1,4 +1,14 @@
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 use std::fmt::Debug;
 //
 // Copyright (c) 2023 ZettaScale Technology
@@ -27,7 +37,17 @@ use futures::{future::select_all, task::AtomicWaker};
 use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 use zenoh_buffers::ZSlice;
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 use zenoh_buffers::{
     buffer::Buffer,
     reader::{BacktrackableReader, HasReader},
@@ -42,7 +62,17 @@ use zenoh_sync::RecyclingObjectPool;
 #[cfg(feature = "unstable")]
 use zenoh_sync::{event, Notifier, Waiter};
 use zenoh_task::TaskController;
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 use zenoh_uring::api::reader::{
     fragmented_batch::{DefragmentationState, FragmentedBatch},
     rx_buffer::RxBuffer,
@@ -370,7 +400,17 @@ async fn rx_task(
     cancellation_token: CancellationToken,
     #[cfg(feature = "stats")] stats: zenoh_stats::LinkStats,
 ) -> ZResult<()> {
-    #[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+    #[cfg(all(
+        feature = "uring",
+        target_os = "linux",
+        any(
+            target_arch = "x86_64",
+            target_arch = "aarch64",
+            target_arch = "riscv64",
+            target_arch = "loongarch64",
+            target_arch = "powerpc64"
+        )
+    ))]
     if transport.manager.state.uring.is_some() && link.link.get_fd().is_ok() {
         return rx_task_uring(
             link,
@@ -571,7 +611,17 @@ impl Drop for TimeoutTracker {
     }
 }
 
-#[cfg(all(feature = "uring", target_os = "linux", any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "riscv64", target_arch = "loongarch64", target_arch = "powerpc64")))]
+#[cfg(all(
+    feature = "uring",
+    target_os = "linux",
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+        target_arch = "loongarch64",
+        target_arch = "powerpc64"
+    )
+))]
 async fn rx_task_uring(
     link: &mut TransportLinkUnicastRx,
     transport: TransportUnicastUniversal,

